modules/data/config.py

```diff
@@ -222,7 +225,9 @@ def __apply_type_validation(self, types: dict, conf: dict):
                             conf[key] = bool(conf[key])
                     elif types[key] == "int":
                         conf[key] = int(conf[key])
-                    elif types[key] == "array":
+                    elif types[key] == "float":
+                        conf[key] = float(conf[key])
```

Comment:
we could use `match` for this if/elif

Reply:
Match is supported by python 3.10+. We still support 3.9 (at least for now)